Add value ptr atomese extension

vsbogd opened the pull request:
Move `PtrValue` implementation to the separate repository.

It contains C++ library:
- opencog/libopencog-neuralnet.so
- opencog/neuralnet/*.h

And Python module:
- opencog.neuralnet

vsbogd committed:
Add PTR_VALUE value type.

Create new library to keep opencog extensions to integrate OpenCog and
deep neural net frameworks. Add new PTR_VALUE to keep pointer to the
native object data in atomspace.

vsbogd committed:
Replace name opencog-tensor by opencog-neuralnet

vsbogd committed:
Simplify includes, use _FOUND instead of HAVE_

vsbogd committed:
Add PtrValue implementation

vsbogd committed:
Add PtrValue unit tests

vsbogd committed:
Cleanup root cmake script

vsbogd committed:
Add cython API for PtrValue

vsbogd committed:
Use opencog.atomspace name to import atomspace module

vsbogd committed:
Set ptrvalue name to the library and import procedure

vsbogd committed:
Add PtrValue Python type constructor

vsbogd committed:
Fix cmake script formatting

vsbogd committed:
Add Python PtrValue unit test

vsbogd committed:
Adding vim temporary files to .gitignore

vsbogd committed:
Replace compiler options tuning by include from cogutil

vsbogd committed:
Rename Python library to neuralnet

vsbogd committed:
Rename Python package to opencog.neuralnet

vsbogd committed:
Fix header define

vsbogd committed:
Align directories with package names

vsbogd committed:
Add opencog-neuralnet library installation

vsbogd committed:
Use atomspace .pxd file installed

vsbogd committed:
Add installation for opencog-neural Python API library

vsbogd committed:
Fix space issues

vsbogd committed:
Remove workaround for Py_INCREF and Py_DECREF

noskill commented on experiments/opencog/opencog-neuralnet/cython/CMakeLists.txt:
```diff
@@ -0,0 +1,22 @@
+INCLUDE_DIRECTORIES(${PYTHON_INCLUDE_DIRS})
+
+# TODO: replace by cogutil include
```
is this file a copy from some cogutil file?

vsbogd replied:
File itself is not a copy, this section is a modified copy and at the moment I am not sure it can be just simply replaced by something from cogutil. Cogutil macros doesn't allow additional include directories which is required here. I believe adding new macro to the cogutil and reusing it there is some separate task.

noskill commented on experiments/opencog/opencog-neuralnet/tests/cython/opencog/test_ptrvalue.py:
```diff
@@ -0,0 +1,27 @@
+    def test_call_grounded_object_predicate(self):
+        obj = TestObject("some object")
+
+        value = PtrValue(obj)
```
I would store value to some ConceptNode to cover common usecase in the test

vsbogd replied:
added unit test in PR https://github.com/singnet/semantic-vision/pull/30

vsbogd committed:
Install opencog-neuralnet to ${PREFIX}/lib

By default ${PREFIX}/lib/opencog libraries are not found by ldconfig. To
make them available lib/opencog folder should be included into
/etc/ld.so.conf. Atomspace libraries work because they are linked by
their installation path.

vsbogd committed:
Build python library on `make test`
